src/devices/badger_orion_me_enc.c

```diff
@@ -0,0 +1,149 @@
+- SS: {32} Serial Number, little-endian value
+- xx: Unknown, values look fixed and depends on the model, could be flags also, battery level too, to be guessed
+- L:{1} Leak
+- xx: other unknown values, flags, model, unit, battery low ? to be guessed.
+- RR: {32} Reading value, scale 10 gallon, little-endian value
+- DD: {32} Daily Reading Value, scale 10 gallon, little-endian value
+- FF:{8} Fixed value, always 0x10
+- CC:{16} CRC-16, poly 0x8005, init 0xFFFF, final XOR 0x0000, from previous 21 bytes.
+- TT: Trailing bits
+
+*/
+static int orion_me_enc_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble_pattern[] = { 0xaa, 0xaa, 0xec, 0x62, 0xec, 0x62};
+
+    uint8_t b[23];
+
+    if (bitbuffer->num_rows > 1) {
+        decoder_logf(decoder, 1, __func__, "Too many rows: %d", bitbuffer->num_rows);
+        return DECODE_FAIL_SANITY;
+    }
+    int msg_len = bitbuffer->bits_per_row[0];
+
+    if (msg_len > 290) {
+        decoder_logf(decoder, 1, __func__, "Packet too long: %d bits", msg_len);
+        return DECODE_ABORT_LENGTH;
+    }
+
+    int offset = bitbuffer_search(bitbuffer, 0, 0, preamble_pattern, sizeof(preamble_pattern) * 8);
+
+    if (offset >= msg_len) {
+        decoder_log(decoder, 1, __func__, "Sync word not found");
+        return DECODE_ABORT_EARLY;
+    }
+
+    if ((msg_len - offset ) < 184 ) {
+        decoder_logf(decoder, 1, __func__, "Packet too short: %d bits", msg_len);
+        return DECODE_ABORT_LENGTH;
+    }
+
+    offset += sizeof(preamble_pattern) * 8;
+    bitbuffer_extract_bytes(bitbuffer, 0, offset, b, 23 * 8);
+
+    // Unwhiten the data coded with IBM Whitening Algorithm LFSR.
+    ibm_whitening(b,23);
+
+    decoder_log_bitrow(decoder, 2, __func__, b, 23 * 8, "Unwhiten MSG");
+
+    if (crc16(b, 23, 0x8005, 0xffff)) {
+        decoder_log(decoder, 1, __func__, "CRC 16 do not match");
+        return DECODE_FAIL_MIC;
+    }
+
+    decoder_log_bitrow(decoder, 2, __func__, b, 23 * 8, "MSG");
+
+    // int msg_len   = b[0]; not used
+    int id            = b[8] << 24 | b[7] << 16 | b[6] << 8 | b[5];
+    int flags_1       = b[9] << 16 | b[10] << 8 | b[11];
+    int leaking       = (b[10] & 0x20) >> 5;
+    int reading_raw   = b[15] << 24 | b[14] << 16 | b[13] << 8 | b[12];
+    int daily_raw     = b[19] << 24 | b[18] << 16 | b[17] << 8 | b[18];
+    int flags_2       = b[20];
+    float volume_gal  = reading_raw * 0.1f; // scale or decimal could defer
+    float dly_vol_gal = daily_raw * 0.1f; // scale or decimal could defer
+
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",             "",                    DATA_STRING, "Orion-MEENC",
+            "id",                "",                    DATA_INT,    id,
+            "leaking",           "Leaking",             DATA_INT,    leaking,
+            "volume_gal",        "Volume-Gallon",       DATA_FORMAT, "%.1f gal",  DATA_DOUBLE, volume_gal,
```

Comment:
Consider calling this "reading" or "current raw meter
reading" -- they way it's referred to in Badger's docs.